packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/config.ts

```diff
@@ -16,7 +16,29 @@ type FeePayerWallet = {
   privateKey: number[]
 }
 
-const readConfig = () => {
+type Config = {
+  endpoint: string
+  discoveryDbConnectionString: string
+  redisUrl: string
+  serverHost: string
+  serverPort: number
+  solanaEndpoints: string[]
+  rewardsManagerProgramId: string
+  rewardsManagerAccountAddress: string
+  claimableTokenProgramId: string
+  paymentRouterProgramId: string
+  trackListenCountProgramId: string
+  usdcMintAddress: string
+  waudioMintAddress: string
+  solanaFeePayerWallets: Keypair[]
+  delegatePrivateKey: Buffer,
+  ipdataApiKey: string | null
+}
+
+let cachedConfig: Config | null = null
+
+const readConfig = (): Config => {
+  if (cachedConfig !== null) return cachedConfig
```

Comment:
is the cache necessary? i dont think anything is too expensive in here, but maybe i dont understand parsing the wallets and that's extra work for now reason

Reply:
it's just to avoid excessive calls

Reply:
sg!